cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -81,28 +82,56 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 
 		// If pool is already present.
 		existingPool, _ := pool.GetPoolName()
-		if common.CheckIfPresent(existingPool, string(pool.PoolPrefix)+string(cStorPoolGot.GetUID())) {
-			common.InitialImportedPoolVol, err = volumereplica.GetVolumes()
-			if err != nil {
-				return string(apis.CStorPoolStatusOffline), err
+		flag := len(existingPool) != 0
+		cnt := common.NoOfPoolWaitAttempts
+		for i := 0; flag && i < cnt; i++ {
+			existingPool, _ := pool.GetPoolName()
+			if common.CheckIfPresent(existingPool, string(pool.PoolPrefix)+string(cStorPoolGot.GetUID())) {
+				flag = true
+				common.InitialImportedPoolVol, err = volumereplica.GetVolumes()
+				if err != nil {
+					common.IsImported <- false
+					return string(apis.CStorPoolStatusOffline), err
+				}
+				if i == cnt-1 {
+					if IsInitStatus(cStorPoolGot) {
+						glog.Infof("Pool %v is online", string(pool.PoolPrefix)+string(cStorPoolGot.GetUID()))
+						c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+						common.IsImported <- true
+						return string(apis.CStorPoolStatusOnline), nil
+					}
+					glog.Errorf("Pool %v already present", string(pool.PoolPrefix)+string(cStorPoolGot.GetUID()))
+					c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+					common.IsImported <- true
+					return string(apis.CStorPoolStatusAlreadyPresent), nil
+				}
+				glog.Infof("Attempt %v: Waiting...", i+1)
+				time.Sleep(common.PoolWaitInterval)
+			} else {
+				flag = false
+				common.InitialImportedPoolVol, err = volumereplica.GetVolumes()
```

Comment:
Lot of logic. Can you please add more comments for better understanding?

Reply:
will address